### PR TITLE
chore(NormalModule.js): Added JSDoc compliant method signatures

### PR DESCRIPTION
**Pull request by jamesgeorge007**
**What kind of change does this PR introduce?**
Meta update.

**Did you add tests for your changes?**
Nope

**Does this PR introduce a breaking change?**
Nope

**What needs to be documented once your changes are merged?**
Nothing

### CODE REVIEW

File: lib/NormalModule.js [hunk 1]
```diff
@@ -673,9 +678,11 @@ class NormalModule extends Module {
 		return rule.test(content);
 	}
 
-	// check if module should not be parsed
-	// returns "true" if the module should !not! be parsed
-	// returns "false" if the module !must! be parsed
+	/**
+	 * @param {string | Function | RegExp | [RegExp, ...RegExp[]] | [string, ...string[]]} noParseRule the parse-rule
```
Comment (sokra) [inline lib/NormalModule.js]: please reference the type generated from schema here

File: lib/NormalModule.js [hunk 2]
```diff
@@ -660,6 +660,11 @@ class NormalModule extends Module {
 		this.addError(error);
 	}
 
+	/**
+	 * @param {string | Function | RegExp} rule the rule
+	 * @param {string} content the content
+	 * @returns {boolean} a boolean or return value from the function
```
Comment (jamesgeorge007) [inline lib/NormalModule.js]: May I know what should be the return type here?  Ig that's gonna be a `boolean` or a value returned by `rule()` if it's a function.

File: lib/NormalModule.js [hunk 3]
```diff
@@ -660,6 +660,11 @@ class NormalModule extends Module {
 		this.addError(error);
 	}
 
+	/**
+	 * @param {string | Function | RegExp} rule the rule
+	 * @param {string} content the content
```
Comment (jamesgeorge007) [inline lib/NormalModule.js]: Can you suggest valid descriptions?

File: lib/NormalModule.js (hunk 1 above)
Comment (jamesgeorge007) [reply lib/NormalModule.js]: Could you please elaborate?

Comment (sokra) [reply lib/NormalModule.js]: please reference the type generated from schema here

File: lib/NormalModule.js [hunk 4]
```diff
@@ -701,6 +708,10 @@ class NormalModule extends Module {
 		return false;
 	}
 
+	/**
+	 * @param {Compilation} compilation the compilation
+	 * @returns {Void} null
```
Comment (sokra) [inline lib/NormalModule.js]:
```suggestion
	 * @returns {void}
```

**Comment by vankop** [comment]
Thanks for your contribution. I am closing PR because it is too stale..

**Pull request closed without merging**